ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -522,6 +525,25 @@ private int executeIncrementalLoad(DriverContext driverContext) {
       // bootstrap of tables if exist.
       if (builder.hasMoreWork() || work.getPathsToCopyIterator().hasNext() || work.hasBootstrapLoadTasks()) {
         DAGTraversal.traverse(childTasks, new AddDependencyToLeaves(TaskFactory.get(work, conf)));
+      } else if (work.dbNameToLoadIn != null) {
+        // Nothing to be done for repl load now. Add a task to update the last.repl.id of the
+        // target database to the event id of the last event considered by the dump. Next
+        // incremental cycle if starts from this id, the events considered for this dump, won't
+        // be considered again. If we are replicating to multiple databases at a time, it's not
+        // possible to know which all databases we are replicating into and hence we can not
+        // update repl id in all those databases.
+        String lastEventid = builder.eventTo().toString();
```

Comment:
Can we try to re-use ReplLoadTask.updateDatabaseLastReplID method instead of duplicating the code here?

Reply:
I think the reason this code is getting duplicated multiple times is the number of variables that change between one code snippet to the other is almost same as the number of lines of code. So, it's probably 50/50 chance that we will have any real benefit from de-duplication.

Reply:
OK